### PR TITLE
[Feature] [Java-api][Catalog]Get all the table information under catalog

### LINKED ISSUE #1176: [Feature] [Java-api][Catalog]Get all the table information under catalog
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Get all the table information under catalog

### Solution

1. Add listTables in `org.apache.paimon.catalog.Catalog` and return `List<Identifier>`
```
List<Identifier>listTables() throws TableNotExistException;
```
2. Implement corresponding logic in `FileSystemCatalog` and `HiveCatalog`

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by liugddx**
### Purpose

close #1176

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java [hunk 1]
```diff
@@ -108,6 +111,31 @@ void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade)
...
     List<String> listTables(String databaseName) throws DatabaseNotExistException;
 
+    /**
+     * Get names of all tables under this catalog. An empty list is returned if none exists.
+     *
+     * <p>NOTE: System tables will not be listed.
+     *
+     * @return a list of the names of all tables in this catalog
+     */
+    default List<Identifier> listTables() {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java:121]: This interface appears to be confused.
List tables from current database or list all tables?

Comment (liugddx) [reply paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java]: > This interface appears to be confused. List tables from current database or list all tables?

I changed it to `listAllTables`, Is this okay?  I want to get all the tables under the catalog.

File: paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java [hunk 2]
```diff
@@ -108,6 +111,31 @@ void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade)
...
     List<String> listTables(String databaseName) throws DatabaseNotExistException;
 
+    /**
+     * Get names of all tables under this catalog. An empty list is returned if none exists.
+     *
+     * <p>NOTE: System tables will not be listed.
+     *
+     * @return a list of the names of all tables in this catalog
+     */
+    default List<Identifier> listAllTables() {
```
Comment (SteNicholas) [inline paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java:121]:
```suggestion
    default List<String> listTables() {
```

**Comment by JingsongLi** [comment]
The usefulness of this method is more difficult to imagine. Just close this.

**Pull request closed without merging**